### ISSUE TITLE
Planet selection zone doesn't adapt

### ISSUE THREAD

**Issue by beccabunny** [endless-sky/endless-sky-editor#32]
Currently, the area from where a planet can be selected is always the same size, which makes it small for most planets, and really small for some (e.g. gas giants)

**Comment by maltedreschert** [comment]
The problem here is that the selection area is tied to the radius value of the planet. The radius values are stored in the StellarObject::INFO map. If a planet type is not in this list, a default radius value of 40.0f is used.
One fix would be to add all the missing planet types in this map (i.e. planet/gas1-b). But I don't know what impact that has on the game.

**Comment by tehhowch** [comment]
The fix is a complete rearchitecting of the editor to not have hardcoded data, but to parse user-supplied data. (This will not affect the actual game.)

I would envision this eventually taking the form of parsing the same directory tree as the map file by default, but allowing the user to specifically supply a path to a resource directory if they choose (via a menu option e.g. "Load Resources").

Data that could be loaded as part of this would include:
 - star sprites
 - planet sprites
 - fleet names
 - commodity names

A much easier solution is to parse a specific file, by default one located in the editor's directory, with an appropriate name such as "stellar-objects.txt", which is at first the same data currently hardcoded into the editor. With this approach I would expect a "load resource file" menu option, to allow the user to load a file of their choice that contains all the data the editor needs for that type of data.

**Comment by maltedreschert** [comment]
I totally agree with you here. It makes total sense to connect it to the game directly. My only question is, how annoying is this bug vs how long does it take to redesign the architecture. Adding the missing values is quickly done but an annoyance to maintain, but if it helps the designers short term it might be worth it.
